### ISSUE TITLE
Does only display the clear color on osx with metal

### ISSUE THREAD

**Issue by Kerollmops** [amethyst/amethyst-starter-2d#1]
Hi,

I tried out to run this starter for 2d games on macos by making care to change the default feature to "metal" and the only thing that it display is the purple color used to clear the screen. According to the code it seems that this program should have displayed variants of the amethyst logo.

**Comment by Kerollmops** [comment]
I changed the `DrawFlat2DTransparentDesc` to not be transparent (i.e. `DrawFlat2DDesc`) and it basically worked.

**Comment by ctsrc** [comment]
I can confirm that the same thing happened when I compiled and ran the amethyst-starter-2d code on a 2018 model MacBook Air with macOS Mojave 10.14.5, using "metal" as the default feature.

With the change described by @Kerollmops above, sprites are shown on screen.

Question is, why doesn't it work with `DrawFlat2DTransparentDesc`?

```diff
diff --git a/src/render.rs b/src/render.rs
index ed0f89a..cccce7f 100644
--- a/src/render.rs
+++ b/src/render.rs
@@ -1,7 +1,7 @@
 use amethyst::{
     ecs::prelude::{ReadExpect, Resources, SystemData},
     renderer::{
-        pass::DrawFlat2DTransparentDesc,
+        pass::DrawFlat2DDesc,
         rendy::{
             factory::Factory,
             graph::{
@@ -76,7 +76,7 @@ impl GraphCreator<DefaultBackend> for RenderGraph {
         let pass = builder.add_node(
             SubpassBuilder::new()
                 // Draw sprites with transparency
-                .with_group(DrawFlat2DTransparentDesc::new().builder())
+                .with_group(DrawFlat2DDesc::new().builder())
                 .with_color(color)
                 .with_depth_stencil(depth)
                 .into_pass(),
```

<img width="912" alt="Screen Shot 2019-07-16 at 16 28 57" src="https://user-images.githubusercontent.com/36199671/61302930-e9e61f00-a7e6-11e9-9114-be1fbb02ed8d.png">

**Comment by happenslol** [comment]
Thanks for your detailed reports!

The images don't contain any transparency - which is probably to be an error on my part. I must've done something stupid, like swapping them out with optimized versions after confirming that they work, and then not re-checking.

However, I've just merged #3, which will make sure that both opaque and transparent sprites are rendered. I'm closing this since it should all be working correctly now.
